lib/views/home/home_view.dart

```diff
@@ -26,11 +28,24 @@ class HomeView extends ConsumerStatefulWidget {
 }
 
 class HomeViewState extends ConsumerState<HomeView> {
-  // TODO: - Network image
-  Image get _backgroundImage => Image(
-        image: Assets.images.nimbleBackground.image().image,
-        fit: BoxFit.cover,
-        alignment: Alignment.center,
+  final _surveyItemController = PageController();
+  final _surveyIndex = ValueNotifier<int>(0);
```

Comment:
I think we don't need to create 2 variables to serve one purpose (set the indicator index). 🤔 Can we set the new index to ViewModel when changing the PageView?

Reply:
@manh-t 

The purpose of the `_surveyIndex` is to set the index to the ViewModel 👉  [here](https://github.com/nmint8m/flutter-survey-app/pull/82/files#diff-5ebf55cf59e234e73423aedda90273d527d7c5f9d96b97aa589258ab156524a4R151-R155)

Technically, I can listen to the `_surveyItemController` and delete the `_surveyIndex`. However, the `page` value which `PageController` provides returns `double` instead of `int`. But if I follow this approach, the listener block will be called every time the user drags the screen. The fact that we only need to listen when the page finishes animation, not every time like that.

Reply:
@nmint8m I think we can create a StateProvider value so we either listen to the value's change or update its value. Something like: 🤔 
```
final _surveyIndex = StateProvider.autoDispose<int>((_) {
  return 0;
});
...
// get the value:
final index = ref.watch(_surveyIndex).state;
// update the value
ref.watch(_surveyIndex).state = 2;
```

Reply:
I tried your suggestion but it didn't work. 🤔

Reply:
@manh-t I created issue to solve it later #86